components/arm/xarm/xarm_comm.go

```diff
@@ -357,9 +357,9 @@ func (x *xArm) MoveToJointPositions(ctx context.Context, newPositions *pb.JointP
 
 	diff := getMaxDiff(from, to)
 	nSteps := int((diff / float64(x.speed)) * x.moveHZ)
```

Comment:
Original problem is actually here; if diff was small enough, the cast to an `int` would result in an `nSteps` of 0 which skipped the entire loop where we issued `MoveJoints` commands.